scripts/md-gen/controller-config/main.go

```diff
@@ -0,0 +1,472 @@
+
+// Get default values using reflection on controller.Config type
+func fillFromNewConfig(data map[string]*keyInfo) {
+	config, err := controller.NewConfig(testing.ControllerTag.Id(), testing.CACert, nil)
+	check(err)
+	for key, defaultVal := range config {
+		if skipDefault.Contains(key) {
+			continue
+		}
+
+		ensureDefined(data, key)
+		data[key].Default = fmt.Sprint(defaultVal)
+	}
+}
+
+// Get default values using reflection on controller.Config type
+// Used as a fallback where fillFromNewConfig can't produce a value
+func fillFromConfigType(data map[string]*keyInfo) {
+	// Don't get defaults from these methods - generally bogus values
+	skipMethods := set.NewStrings(
+		"CAASImageRepo",
+		"CAASOperatorImagePath",
+		"ControllerAPIPort",
+		"Features",
+		"IdentityPublicKey",
+		"Validate", // not a config key
+	)
+
+	constantNameForMethod := func(methodName string) string {
+		name := strings.TrimSuffix(methodName, "MB")
+
+		rename := map[string]string{
+			"NUMACtlPreference": "SetNUMAControlPolicyKey",
+		}
+		if rn, ok := rename[name]; ok {
+			name = rn
+		}
+
+		return name
+	}
+
+	config, err := controller.NewConfig(testing.ControllerTag.Id(), testing.CACert, nil)
+	check(err)
+	t := reflect.TypeOf(config)
+	v := reflect.ValueOf(config)
+
+	for i := 0; i < t.NumMethod(); i++ {
+		method := t.Method(i)
+		methodValue := v.Method(i)
+
+		if skipMethods.Contains(method.Name) {
+			continue
+		}
+		if method.Type.NumIn() == 1 {
+			defaultVal := methodValue.Call([]reflect.Value{})[0]
+
+			constantName := constantNameForMethod(method.Name)
+			key, ok := keyForConstantName[constantName]
+			if !ok {
+				// Try adding "Key" suffix
+				key, ok = keyForConstantName[constantName+"Key"]
+				if !ok {
+					panic(method.Name)
+				}
+			}
+			if skipDefault.Contains(key) {
+				continue
+			}
+
+			ensureDefined(data, key)
+			data[key].Default2 = fmt.Sprint(defaultVal)
+		}
+	}
+}
+
+// UTILITY FUNCTIONS
+
+// Return the first value that is defined / not-zero, and "true"
+// if such a value is found.
+func firstNonzero(vals ...string) (string, bool) {
+	for _, val := range vals {
+		if val != "" {
+			return val, true
+		}
+	}
+	return "", false
+}
+
+// Ensure that the data map has an entry for key.
+func ensureDefined(data map[string]*keyInfo, key string) {
+	if data[key] == nil {
+		data[key] = &keyInfo{
+			Key: key,
+		}
+	}
+}
+
+// check panics if the provided error is nil.
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// return keys of the given map in descending length order
+func getKeysInDescLenOrder[T any](m map[string]T) (keys []string) {
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return len(keys[i]) > len(keys[j])
+	})
+	return
+}
+
+// Dump a YAML representation of v to the given file
+func dumpYAML(path string, v any) {
+	file, err := os.Create(path)
+	check(err)
+	yamlEncoder := yaml.NewEncoder(file)
+	check(yamlEncoder.Encode(v))
+}
+
+// UNUSED
```

Comment:
IMO we should leave these out of what's merged (but they'll still be accessible in the PR), so that we don't have dead code in our repo.